agents/tester/main.py

```diff
@@ -0,0 +1,41 @@
+import fixieai
+
+# This is an Agent that can be used to test other Agents. Given an agent name, a query, and an expected result,
+# the Agent calls out to the requested Agent-under-test, gets back its response, and compares the response to
+# the expected result. If the response matches the expected result, the test passes. Otherwise, the test fails.
+
+BASE_PROMPT = """I am an agent that can be used to test other Agents."""
+FEW_SHOTS = """
+Q: Test fixie/calc with the query: "What is 45 * 72?". The expected answer is "3240".
+Ask Agent[fixie/calc]: "What is 45*72"?
+Agent[fixie/calc] says: 3240
+Thought: I need to compare the result to the expected answer.
+Thought: Is the result "3240" structurally equivalent to the expected answer "3240"?
+Thought: The result matches the expected answer exactly. Therefore, the test passed.
+A: The fixie/calc agent responded with "3240" and we expected "3240". The test passed.
+
+Q: Test fixie/liveinfo with the query: "What is the weather in Boston today?". The expected answer is "The current weather in Boston, MA is 57 degrees and sunny."
```

Comment:
I think this should probably be more along the lines of "The expected answer must contain a weather forecast for Boston.", or else it will be extremely fragile.

Reply:
The point of these few-shots is just to give the agent the pattern of "When you are asked to test agent X with query Q and expected answer A, here is what you do." The specific details of the "tests" in these few shots don't really matter as long as the LLM does the right thing with expanding the real tests. I can change it if you insist but I don't think it will end up mattering since these are not the *actual* test cases.